array-methods-task3.js

```diff
@@ -0,0 +1,25 @@
+// Filter range "in place"
+// importance: 4
+
+// Write a function filterRangeInPlace(arr, a, b) that gets an array arr and removes from it all values except those that are between a and b. The test is: a ≤ arr[i] ≤ b.
+
+// The function should only modify the array. It should not return anything.
+
+// For instance:
+
+let arr = [5, 3, 8, 1];
+
+filterRangeInPlace(arr, 1, 4);
+// removed the numbers except from 1 to 4
+
+// alert( arr ); // [3, 1]
+function filterRangeInPlace(arr, a, b) {
+  for (let i = 0, j = 0; i < arr.length, j < arr.length; i++, j++) {
+    if (arr[i] > b || arr[j] < a) {
+      arr.splice(i, 1);
+      i--, j--;
+    }
+  }
```

Comment:
i and j are doing exactly the same, you can remove `j`

Reply:
![2e8syk](https://github.com/user-attachments/assets/739f4e37-2280-4899-9795-041e4d630417)